haystack_experimental/dataclasses/chat_message.py

```diff
@@ -0,0 +1,187 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import base64
+from dataclasses import asdict, dataclass, field
+from typing import Any, Dict, List, Literal, Optional, Sequence, Union
+
+import filetype
+import haystack.dataclasses.chat_message
+from haystack import logging
+from haystack.dataclasses import ChatMessage as HaystackChatMessage
+from haystack.dataclasses import ChatRole, TextContent, ToolCall, ToolCallResult
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ImageContent:
+    """
+    The image content of a chat message.
+
+    :param base64_image: A base64 string representing the image.
+    :param mime_type: The mime type of the image.
+    :param detail: Optional detail level of the image (only supported by OpenAI). One of "auto", "high", or "low".
+    :param meta: Optional metadata for the image.
+    """
+
+    base64_image: str
+    mime_type: Optional[str] = None
+    detail: Optional[Literal["auto", "high", "low"]] = None
+    meta: Dict[str, Any] = field(default_factory=dict)
+
+    def __post_init__(self):
+        # mime_type is an important information, so we try to guess it if not provided
+        if not self.mime_type:
+            try:
+                # Attempt to decode the string as base64
+                decoded_image = base64.b64decode(self.base64_image)
+
+                guess = filetype.guess(decoded_image)
+                if guess:
+                    self.mime_type = guess.mime
```

Comment:
Do you think it'd make sense to add a logger message here the in situation no mimetype was found?

Reply:
maybe here it makes more sense, since it will be emitted only if the base64 image is valid and we cannot guess this relevant information. I'll add a warning.

Reply:
https://github.com/deepset-ai/haystack-experimental/pull/286/commits/e1eafc3a7fd98727f6f0109b4dd6bce263813d9f